packages/oui-switch/src/switch.controller.js

```diff
@@ -0,0 +1,18 @@
+import { addBooleanParameter } from '@oui-angular/common/component-utils'
+
+export default class {
+  constructor ($scope, $attrs, $transclude) {
+    'ngInject'
+
+    this.$scope = $scope
```

Comment:
What do you think about avoiding using $scope ?

I hope one day, we will just do a breaking release and pass all our component to ng4+ :)

Reply:
We should have a look to : https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#controlleras-controller-syntax

Reply:
It's just an easy way to create a unique id base on the $id property of scope here.
Thank you for the doc I discover now all these rules (I'm kidding), unfortunately, avoiding using $scope in a huge components library ... and directives (yes, sorry) library in angular < 2 is a pure utopia.

Reply:
I strongly disagree with doc before code. If you want down to earth API / interface it's better to implement things based on a real life needs and not on some sort of documentation.

Remember, this is a component. We can introduce change easily if we choose (or need) to. I think the component in it's current form encompass what we need for step 1 => A dropdown for our action menu. I'm not saying we should forget about the future, however, we need to stay away from over-design.